multi-node/generic/controller-install.sh

```diff
@@ -946,16 +818,21 @@ function start_addons {
     do
         sleep 5
     done
-    echo
-    echo "K8S: DNS addon"
-    curl --silent -H "Content-Type: application/json" -XPOST -d"$(cat /srv/kubernetes/manifests/kube-dns-rc.json)" "http://127.0.0.1:8080/api/v1/namespaces/kube-system/replicationcontrollers" > /dev/null
-    curl --silent -H "Content-Type: application/json" -XPOST -d"$(cat /srv/kubernetes/manifests/kube-dns-svc.json)" "http://127.0.0.1:8080/api/v1/namespaces/kube-system/services" > /dev/null
-    echo "K8S: Heapster addon"
-    curl --silent -H "Content-Type: application/json" -XPOST -d"$(cat /srv/kubernetes/manifests/heapster-de.json)" "http://127.0.0.1:8080/apis/extensions/v1beta1/namespaces/kube-system/deployments" > /dev/null
-    curl --silent -H "Content-Type: application/json" -XPOST -d"$(cat /srv/kubernetes/manifests/heapster-svc.json)" "http://127.0.0.1:8080/api/v1/namespaces/kube-system/services" > /dev/null
-    echo "K8S: Dashboard addon"
-    curl --silent -H "Content-Type: application/json" -XPOST -d"$(cat /srv/kubernetes/manifests/kube-dashboard-rc.json)" "http://127.0.0.1:8080/api/v1/namespaces/kube-system/replicationcontrollers" > /dev/null
-    curl --silent -H "Content-Type: application/json" -XPOST -d"$(cat /srv/kubernetes/manifests/kube-dashboard-svc.json)" "http://127.0.0.1:8080/api/v1/namespaces/kube-system/services" > /dev/null
+
+    for f in /srv/kubernetes/addons/{kube-{dns-{rc,svc},dashboard-{rc,svc}},heapster-{de,svc}}.yaml;do
+      uuid_file_save="/var/run/coreos/rkt-kubectl-${RANDOM}.uuid"
```

Comment:
Would be cleaner to move this into a function then just call function for each path. And just an "easier to maintain" nit would be using explicit paths:

```
create-addon /srv/kubernetes/addons/kube-dns-rc.yaml
create-addon /srv/kubernetes/addons/kube-dns-svc.yaml
...
```